internal/server/data/migrator/helpers.go

```diff
@@ -0,0 +1,57 @@
+package migrator
+
+import "gorm.io/gorm"
+
+type Tx interface {
+	Exec(stmt string, args ...any) *gorm.DB
+}
+
+// HasTable returns true if the database has a table with name. Returns
+// false if the table does not exist, or if there was a failure querying the
+// database.
+func HasTable(tx *gorm.DB, name string) bool {
+	var count int
+	stmt := `
+		SELECT count(*)
+		FROM information_schema.tables
+		WHERE table_schema = CURRENT_SCHEMA()
+		AND table_name = ? AND table_type = 'BASE TABLE'
+	`
+	if tx.Dialector.Name() == "sqlite" {
+		stmt = `SELECT count(*) FROM sqlite_master WHERE type = 'table' AND name = ?`
+	}
+
+	if err := tx.Raw(stmt, name).Scan(&count).Error; err != nil {
+		return false
+	}
+	return count != 0
+}
+
+// HasColumn returns true if the database table has the column. Returns false if
+// the database table does not have the column, or if there was a failure querying
+// the database.
+func HasColumn(tx *gorm.DB, table string, column string) bool {
+	var count int
+
+	stmt := `
+		SELECT count(*)
+		FROM information_schema.columns
+		WHERE table_schema = CURRENT_SCHEMA()
+		AND table_name = ? AND column_name = ?
+	`
+
+	if tx.Dialector.Name() == "sqlite" {
+		stmt = `
+			SELECT count(*)
+			FROM sqlite_master
+			WHERE type = 'table' AND name = ?
+			AND sql LIKE ?
+		`
+		column = "%`" + column + "`%"
```

Comment:
I based this on the gorm migrator implementation we're using now: https://github.com/go-gorm/sqlite/blob/v1.3.6/migrator.go#L70-L71

The `sqlite_master` table stores the `CREATE TABLE` statement as sql, so we have to use substring matching to find the column.

The gorm implementation has a lot more `sql LIKE ?` parts of the query. My guess is that different sqlite implementations might use different quoting for column names. This works for our existing use cases, and since we won't support sqlite in production, this seems sufficient for our use case.